pkg/coscheduling/coscheduling.go

```diff
@@ -70,17 +66,7 @@ func New(obj runtime.Object, handle framework.Handle) (framework.Plugin, error)
 	pgClient := pgclientset.NewForConfigOrDie(handle.KubeConfig())
 	pgInformerFactory := pgformers.NewSharedInformerFactory(pgClient, 0)
 	pgInformer := pgInformerFactory.Scheduling().V1alpha1().PodGroups()
-
-	fieldSelector, err := fields.ParseSelector(",status.phase!=" + string(v1.PodSucceeded) + ",status.phase!=" + string(v1.PodFailed))
```

Comment:
We don't need to filter out the pods in succeeded and failed phase any more?

Reply:
The informer carried over from the scheduler framework has already done this.

Reply:
Correct me if I am wrong: if we filter out succeeded and failed pods then we will potentially not allow scheduling of pods that are still pending in case some pods succeeded or failed earlier because number of pods will be smaller than minAvailable.

Reply:
This is why we're relying on the "snaphost" instead of podLister to calculate the number of "available" pods:

https://github.com/kubernetes-sigs/scheduler-plugins/blob/417f2f6ea22035e72e8afb095b9af5d236427137/pkg/coscheduling/core/core.go#L267-L269

(Note that assigned and assumed pods can be obtained from the "snapshot").

Reply:
For now, coscheduling obtained the assigned + assumed pods from scheduler's snapshot. In this case, succeeded & failed pods will be both excluded. As you mentioned, this might be problematic, in particular with excluded succeeded pods.

According to your exp., how practical this problem may exist in a real-world env?

Reply:
I think it depends entirely on how the operator/controller handles failed pods. It may recreate pods after some pods fail. If we don't exclude failed or succeeded pods. These new pods will always meet the minimum number limit in the co-scheduling. But there may not be enough resources in the cluster to allow all newly created pods to be scheduled. So we need to exclude the failed or successful pods as well.

Reply:
I think we should include succeeded and failed pods when comparing against `.spec.minAvailable` in Permit. 

Imagine scenario with a job where every pod needs to succeed. If one of the pods is preempted and other pods from the same group succeeded, recreated pod will never be scheduled, but it should pass `minAvailable` check in Permit. Only solution for this would be some kind of custom control mechanism updating pod group resource which is pretty complicated from user perspective.

> But there may not be enough resources in the cluster to allow all newly created pods to be scheduled. So we need to exclude the failed or successful pods as well.

I do not understand this comment. Are you talking whether to include or exclude finished pods in PreFilter check regarding resources available in the cluster?

What exactly are the benefits of ignoring succeed/failed pods?

Reply:
Anyways, we should not be blocking the review on this, we can move this discussion to an issue.

Reply:
I'd like to pursue embedding a non-filtered indexInformer in the framework handle, and this has to be implemented in the upstream.

Reply:
It seems in sharedInformerFactory, you cannot associate two informers for one API resource:

https://github.com/kubernetes/kubernetes/blob/9a75e7b0fd1b567f774a3373be640e19b33e7ef1/staging/src/k8s.io/client-go/informers/factory.go#L179-L180

Reply:
>  If one of the pods is preempted and other pods from the same group succeeded, recreated pod will never be scheduled, but it should pass minAvailable check in Permit.

Please correct me if I have misrepresented. @mateuszlitwin  @Huang-Wei 

Does this scenario mean that the job can still run even if some workers are preempted or fail?  Or can a newly created Pod continue to get the job done without having to cooperate with other Pods(other succeed pods)?

I think we should be clear about the concepts expressed in the minAvailable. 
1. the minimum runnable instance that satisfies the job run.
2. or succeed + failed + running？

Reply:
> Anyways, we should not be blocking the review on this, we can move this discussion to an issue.
+1

Reply:
Let me give an example. A Job requires 3 replicas to run altogether. And `Pod1`, `Pod2`, `Pod3` all get scheduled and running. Now let's say `Pod1` and `Pod2` succeeded, but `Pod3` failed. Then the controller spins up a replacement pod `Pod3'`. In this case, `Pod3'` wouldn't be scheduled as `Pod1` and `Pod3` have been excluded from the scheduler snapshot as well as the "filtered" podinformer.

@mateuszlitwin I suppose this is your concern, right?

Reply:
Yes. If Pod3 is deleted (e.g. Pod3 was preempted due to node going down or was lower priority than other pod) and then recreated it will not schedule.